lib/matrix.gd

```diff
@@ -2019,6 +2019,78 @@ DeclareOperation("DefaultScalarDomainOfMatrixList",[IsListOrCollection]);
 ##
 DeclareOperation("BaseField",[IsObject]);
 
+#############################################################################
+##
+#O  RationalCanonicalFormTransform( <mat> )
+##
+##  <#GAPDoc Label="RationalCanonicalFormTransform">
+##  <ManSection>
+##  <Func Name="RationalCanonicalFormTransform" Arg='mat'/>
+##
+##  <Description>
+##  For a matrix <A>A</A> return a matrix <A>P</A> such that 
+##  <M><A>A</A>^<A>P</A></M> is in rational canonical form (also called
+##  Frobenius normal form). The algorithm used is the basic textbook
+##  version and thus not of optimal complexity.
+##  <Example><![CDATA[
+##  gap> aa:=[[0,-8,12,40,-36,4,0,59,15,-9],[-2,-2,-2,6,-11,1,-1,10,1,0],
+##  > [1,5,0,-6,12,-2,0,-12,-4,2],[0,0,0,2,0,0,0,7,0,0],
+##  > [0,2,-3,-7,8,-1,0,-7,-3,2],[-5,-4,-6,18,-30,2,-2,35,5,-1],
+##  > [-1,-6,6,20,-28,3,0,24,10,-6],[0,0,0,-1,0,0,0,-3,0,0],
+##  > [0,0,-1,-2,-2,0,-1,-7,0,0],[0,-8,9,21,-36,4,-2,12,12,-8]];;
+##  gap> t:=RationalCanonicalFormTransform(aa);;
+##  gap> aa^t;
+##  [ [ 0, 0, 0, 1, 0, 0, 0, 0, 0, 0 ], [ 1, 0, 0, 0, 0, 0, 0, 0, 0, 0 ],
+##    [ 0, 1, 0, 0, 0, 0, 0, 0, 0, 0 ], [ 0, 0, 1, 0, 0, 0, 0, 0, 0, 0 ],
+##    [ 0, 0, 0, 0, 0, 0, 0, 0, 0, 1 ], [ 0, 0, 0, 0, 1, 0, 0, 0, 0, 1 ],
+##    [ 0, 0, 0, 0, 0, 1, 0, 0, 0, 1 ], [ 0, 0, 0, 0, 0, 0, 1, 0, 0, 0 ],
+##    [ 0, 0, 0, 0, 0, 0, 0, 1, 0, -1 ], [ 0, 0, 0, 0, 0, 0, 0, 0, 1, -1 ] ]
+##  ]]></Example>
+##  </Description>
+##  </ManSection>
+##  <#/GAPDoc>
+##
+DeclareGlobalFunction( "RationalCanonicalFormTransform" );
+
+#############################################################################
+##
+#O  SimplexMethod( <A>, <b>, <c> )
+##
+##  <#GAPDoc Label="SimplexMethod">
+##  <ManSection>
+##  <Oper Name="SimplexMethod" Arg='A,b,c'/>
+##
+##  <Description>
+##  Find a rational vector <A>x</A> that maximizes <M><A>x</A>\cdot<A>c</A></M>, subject
+##  to the constraint <M><A>A</A>\cdot<A>x</A>\le<A>b</A></M>. 
+##  <Example><![CDATA[
+##  gap> A:=[[3,1,1,4],[1,-3,2,3],[2,1,3,-1]];;
+##  gap> b:=[12,7,10];;c:=[2,4,3,1];;
+##  gap> SimplexMethod(A,b,c);
+##  [ [ 0, 52/5, 0, 2/5 ], 42 ]
+##  ]]></Example>
+##  </Description>
+##  </ManSection>
+##  <#/GAPDoc>
+##
+DeclareGlobalFunction( "SimplexMethod" );
```

Comment:
Is there a deeper reason that this manual entries were moved? It makes it very difficult to see if you also made changes to the text, so if possible, it would be nice if they could be moved back to their original position a few lines down.